src/v/raft/types.cc

```diff
@@ -553,6 +553,51 @@ void heartbeat_reply::serde_read(iobuf_parser& src, const serde::header& hdr) {
     }
 }
 
+ss::future<> append_entries_request::serde_async_write(iobuf& dst) {
+    auto mem_batches = co_await model::consume_reader_to_memory(
+      std::move(batches()), model::no_timeout);
+
+    iobuf out;
+    using serde::write;
+
+    write(out, static_cast<uint32_t>(mem_batches.size()));
```

Comment:
maybe we can use `async_adl` for either record batch reader or  circular buffer ?

Reply:
yield after reflection serialize in lop